bugwarrior/services/todoist.py

```diff
@@ -0,0 +1,286 @@
+    # this is a workaround for https://github.com/ralphbean/taskw/issues/172
+    def _unescape_content(self, content):
+        return (
+            content.replace('"', "'")  # prevent &dquote; in task details
+            .replace("[", self.config.char_open_bracket)  # prevent &open; and &close;
+            .replace("]", self.config.char_close_bracket)
+        )
+
+    def to_taskwarrior(self):
+        default_time = time(0, 0, 0)
+        # adjust timezone to use local time for "floating" dates
+        if self.record["due"]:
+            # The Todoist due date could be a `date` or `datetime`
+            if isinstance(self.record["due"]["date"], datetime):
+                if self.record["due"]["timezone"]:
+                    todoist_due = self.record["due"]["date"]
+                else:
+                    # if no timezone set is set remove tzinfo
+                    # otherwixe it will be treated as UTC by default
+                    todoist_due = self.record["due"]["date"].replace(tzinfo=None)
+            else:
+                # the due is just a `date` with no time or timezone.
+                todoist_due = datetime.combine(
+                    self.record["due"]["date"], default_time, tzinfo=None
+                )
+        else:
+            todoist_due = None
+
+        # deadline if set is only a date with no time or timezone.
+        todoist_deadline = (
+            datetime.combine(self.record["deadline"]["date"], default_time, tzinfo=None)
+            if self.record["deadline"]
+            else None
+        )
+
+        task = {
+            "project": self.extra["project"],
+            "priority": self.get_priority(),
+            "annotations": self.extra.get("annotations", []),
+            "tags": (
+                self.get_tags_from_labels(self.record["labels"])
+                if self.record["labels"]
+                else []
+            ),
+            "scheduled": None,
+            "due": todoist_due,
+            "status": "completed" if self.record["is_completed"] else "pending",
+            "entry": self.record["created_at"],
+            self.ID: self.record["id"],
+            self.CONTENT: self._unescape_content(self.record["content"]),
+            self.DESCRIPTION: self._unescape_content(self.record["description"]),
+            self.DUE: todoist_due,
+            self.DEADLINE: todoist_deadline,
+            self.DURATION: self.extra["duration"],
+            self.ASSIGNEE: self.extra["assignee"],
+            self.ASSIGNER: self.extra["assigner"],
+            self.SECTION: self.extra["section"],
+            self.URL: self.record["url"],
+        }
+        return task
+
+    def get_default_description(self):
+        description = self.build_default_description(
+            title=self._unescape_content(self.record["content"]),
+            url=self.record["url"],
+            number=self.record["id"],
+            cls="issue",
```

Comment:
```suggestion
            cls="task",
```

This will remove the "Is" from the description which probably doesn't make sense for this service.

Reply:
Never realized there was different task types. I'll look at adding proper `subtask` support as well as Todoist has subtasks.

Should Logseq also change the task type from issue to `task` or `todo`?

Reply:
> Should Logseq also change the task type from issue to `task` or `todo`?

Yeah, I suspect both services are better aligned with `task`.